src/colorizer/ColorRamp.ts

```diff
@@ -97,7 +97,13 @@ export default class ColorRamp {
     // Scale t so it represents a (float) index in the array of color stops
     const tIndex = t * (this.colorStops.length - 1);
 
-    // Get the two colors on either side of the tIndex
+    if (this.type === ColorRampType.CATEGORICAL) {
+      // For categorical ramps, we return the nearest color
+      return this.colorStops[Math.round(tIndex)].clone();
+    }
```

Comment:
Fixes a bug where `colorRamp.sample()` would not reflect the color map interpolation settings.